overlord/snapshotstate/snapshotmgr.go

```diff
@@ -73,7 +73,7 @@ func Manager(st *state.State, runner *state.TaskRunner) *SnapshotManager {
 	runner.AddHandler("forget-snapshot", doForget, nil)
 	runner.AddHandler("check-snapshot", doCheck, nil)
 	runner.AddHandler("restore-snapshot", doRestore, undoRestore)
-	runner.AddCleanup("restore-snapshot", cleanupRestore)
+	runner.AddHandler("cleanup-restore-snapshot", doCleanup, nil)
```

Comment:
this takes care of multiple snapshots so it's not clear if it should follow the naming of the other tasks, so maybe:

cleanup-after-restore and doCleanupAfterRestore

?